src/Workspaces/CoreTestUtilities/Fakes/SimpleAssetSource.cs

```diff
@@ -19,9 +19,8 @@ namespace Microsoft.CodeAnalysis.Remote.Testing;
 internal sealed class SimpleAssetSource(ISerializerService serializerService, IReadOnlyDictionary<Checksum, object> map) : IAssetSource
 {
     public ValueTask GetAssetsAsync<T, TArg>(
-        Checksum solutionChecksum, AssetPath assetPath, ReadOnlyMemory<Checksum> checksums, ISerializerService deserializerService, Action<int, T, TArg> callback, TArg arg, CancellationToken cancellationToken)
+        Checksum solutionChecksum, AssetPath assetPath, ReadOnlyMemory<Checksum> checksums, ISerializerService deserializerService, Action<Checksum, T, TArg> callback, TArg arg, CancellationToken cancellationToken)
```

Comment:
because we can find results in any order, we pass back the checksum we found when we find it, not the index in passed-in checksums buffer.  This actually cleans things up in many layers.